services/tasks/local.go

```diff
@@ -664,7 +664,7 @@ func (l *local) writeContent(ctx context.Context, mediaType, ref string, r io.Re
 	if err != nil {
 		return nil, err
 	}
-	if err := writer.Commit(ctx, 0, ""); err != nil {
+	if err := writer.Commit(ctx, 0, ""); err != nil && !errdefs.IsAlreadyExists(err) {
```

Comment:
It's safe to skip ErrAlreadyExist, because

* One container can be checkpointed multiple times. It is safe to ignore ErrAlreadyExist.
* If task doesn't change, the checkpoint content is the same. It's safe to ignore ErrAlreadyExist.


NOTE: The following code ignores ErrAlreadyExists. I think we might rethink the `Checkpoint` API which should only commit the checkpoint content. The container spec should be handled in client side. It might be flexible.

https://github.com/containerd/containerd/blob/8eea15f891099035f29088413cd269c7d8fd4e45/container.go#L373